pkg/querier/vcs/source/find_go.go

```diff
@@ -0,0 +1,142 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+
+	vcsv1 "github.com/grafana/pyroscope/api/gen/proto/go/vcs/v1"
+	"github.com/grafana/pyroscope/pkg/querier/vcs/client"
+	"github.com/grafana/pyroscope/pkg/querier/vcs/source/golang"
+)
+
+const (
+	ExtGo = ".go"
+)
+
+// findGoFile finds a go file in a vcs repository.
+func (ff FileFinder) findGoFile(ctx context.Context) (*vcsv1.GetFileResponse, error) {
+	if url, ok := golang.StandardLibraryURL(ff.path); ok {
+		return ff.fetchURL(ctx, url, false)
+	}
+
+	if relativePath, ok := golang.VendorRelativePath(ff.path); ok {
+		return ff.fetchRepoFile(ctx, relativePath, ff.ref)
+	}
+
+	modFile, ok := golang.ParseModulePath(ff.path)
+	if ok {
+		mainModule := module.Version{
+			Path:    filepath.Join(ff.repo.GetHostName(), ff.repo.GetOwnerName(), ff.repo.GetRepoName()),
```

Comment:
Is `filepath.Join` intentional here? This will use `\` on Windows. I'm not sure if we care about Windows or not.

If we always want `/`, then `path.Join` is a better option.

Reply:
ho yeah good catch !